src/TypeScript.UnitTests/NodeJSLocator/Locators/GlobalPathNodeLocatorTests.cs

```diff
@@ -0,0 +1,101 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator.Locators;
+
+namespace SonarLint.VisualStudio.TypeScript.UnitTests.NodeJSLocator.Locators
+{
+    [TestClass]
+    public class GlobalPathNodeLocatorTests
+    {
+        private readonly string programFilesPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "nodejs\\node.exe");
+
+        [TestMethod]
+        public void Locate_FileNotFound_Null()
+        {
+            using var scope = CreateEnvironmentVariableScope();
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(It.IsAny<string>())).Returns(false);
+
+            var testSubject = CreateTestSubject(fileSystem.Object);
+            var result = testSubject.Locate();
+
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Locate_FileFoundInPath_FilePath()
+        {
+            const string filePath = "c:\\test\\node.exe";
+            using var scope = CreateEnvironmentVariableScope(Path.GetDirectoryName(filePath));
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(filePath)).Returns(true);
+
+            var testSubject = CreateTestSubject(fileSystem.Object);
+            var result = testSubject.Locate();
+
+            result.Should().Be(filePath);
+
+            fileSystem.VerifyAll();
+            fileSystem.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void Locate_FileNotFoundInPath_FileFoundInProgramFiles_FilePath()
+        {
+            using var scope = CreateEnvironmentVariableScope();
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(programFilesPath)).Returns(true);
+
+            var testSubject = CreateTestSubject(fileSystem.Object);
+            var result = testSubject.Locate();
+
+            result.Should().Be(programFilesPath);
+
+            fileSystem.Verify(x=> x.File.Exists(programFilesPath), Times.Once);
+        }
+
+        private GlobalPathNodeLocator CreateTestSubject(IFileSystem fileSystem)
+        {
+            return new GlobalPathNodeLocator(fileSystem, Mock.Of<ILogger>());
+        }
+
+        private EnvironmentVariableScope CreateEnvironmentVariableScope(string path = null)
+        {
+            var scope = new EnvironmentVariableScope();
+
+            // remove any existing node.exe from machine's PATH for testing purposes
+            scope.SetPath(path ?? " ");
```

Comment:
Empty environment variables are normally returned as empty strings, aren't they?
Why is the "empty" path a non-empty string?

Reply:
No, they're returned as [nulls](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=net-5.0#System_Environment_GetEnvironmentVariable_System_String_).  So the product code fails on the `.Split` call, and since the path is unlikely to be null in reality, I preferred to tweak the test rather than the product code.

Reply:
I'll change the product code and add the null check. It's probably safer anyway.